src/froidure-pin.cpp

```diff
@@ -739,14 +740,16 @@ elements are sorted, or :any:`UNDEFINED` if *i* is greater than
     bind_froidure_pin<Perm<0, uint8_t>>(m, "Perm1");
     bind_froidure_pin<Perm<0, uint16_t>>(m, "Perm2");
     bind_froidure_pin<Perm<0, uint32_t>>(m, "Perm4");
-    // TODO(0) uncomment
-    // bind_froidure_pin<detail::KBE,
-    //                   FroidurePinTraits<detail::KBE,
-    //                   fpsemigroup::KnuthBendix>>(
-    //     m, "KBE");
-    // bind_froidure_pin<detail::TCE,
-    //                   FroidurePinTraits<detail::TCE, detail::TCE::Table>>(
-    //     m, "TCE");
+    // TODO(0) Something for Congruence?
+    bind_froidure_pin<
+        detail::KBE<detail::KnuthBendixImpl<detail::RewriteFromLeft>>>(
+        m, "KBERewriteFromLeft");
+    bind_froidure_pin<
+        detail::KBE<detail::KnuthBendixImpl<detail::RewriteTrie>>>(
+        m, "KBERewriteTrie");
+    bind_froidure_pin<detail::TCE>(m, "TCE");
+    bind_froidure_pin<detail::KE<std::string>>(m, "KEString");
+    bind_froidure_pin<detail::KE<word_type>>(m, "KEWord");
```

Comment:
Are these names okay? Happy to receive suggestions.

Reply:
I don't think these are required at all

Reply:
Without them we can't convert the return type of `to` into a Python type.

Reply:
Ah wait I misunderstood the purpose of this, yes, I'm happy with the names! Sorry